eureka-core/src/main/java/com/netflix/eureka/util/AwsAsgUtil.java

```diff
@@ -142,7 +152,13 @@ public void setStatus(String asgName, boolean enabled) {
      * @return - true if the ASG is disabled, false otherwise
      */
     private boolean isAddToLoadBalancerSuspended(String asgName) {
-        AutoScalingGroup asg = retrieveAutoScalingGroup(asgName);
+        String asgAccount = getASGAccount(asgName);
+        AutoScalingGroup asg;
+        if(asgAccount.equals(accountId)) {
```

Comment:
The accountId metadata field was added to eureka-client relatively recently, so for backwards compatibility, it is possible in existing deployments to have older clients that do not register this field. This will cause an NPE at this point in code. Can you please add a check for null asgAccount, and in the null case treat this as same account as the server?